test/translator/read_capacity_10_test.cc

```
@@ -0,0 +1,170 @@
+  }
+
+  // You can define per-test set-up logic as usual.
+  virtual void SetUp() {
+    identify_ns_ = {};
+    SetNamespace(&identify_ns_);
+
+    read_capacity_10_cmd_ = {};
+    SetCommand();
+    nvme_cmds_ = identify_cmds_;
+    memset(buffer_, 0, sizeof(buffer_));
+  }
+
+  void SetCommand() {
+    const uint8_t* cmd_ptr =
+        reinterpret_cast<const uint8_t*>(&read_capacity_10_cmd_);
+    scsi_cmd_ = absl::MakeSpan(cmd_ptr, sizeof(scsi::ReadCapacity10Command));
+  }
+
+  void SetNamespace(nvme::IdentifyNamespace* identify_ns_) {
+    identify_cmds_[0].dptr.prp.prp1 = reinterpret_cast<uint64_t>(identify_ns_);
+  }
+};
+
+TEST_F(ReadCapacity10Test, ToNvmeSuccess) {
+  translator::Allocation allocation{};
+  EXPECT_EQ(translator::ReadCapacity10ToNvme(scsi_cmd_, identify_cmds_[0], 1,
+                                             allocation),
+            translator::StatusCode::kSuccess);
+}
+
+TEST_F(ReadCapacity10Test, ToNvmeBadBuffer) {
+  uint8_t bad_buffer[1];
+  translator::Allocation allocation{};
+  EXPECT_EQ(translator::ReadCapacity10ToNvme(bad_buffer, identify_cmds_[0], 1,
+                                             allocation),
+            translator::StatusCode::kInvalidInput);
+}
+
+TEST_F(ReadCapacity10Test, ToNvmeBadControlByteNaca) {
+  read_capacity_10_cmd_.control_byte.naca = 1;
+  translator::Allocation allocation{};
+  EXPECT_EQ(translator::ReadCapacity10ToNvme(scsi_cmd_, identify_cmds_[0], 1,
+                                             allocation),
+            translator::StatusCode::kInvalidInput);
+}
+
+TEST_F(ReadCapacity10Test, Success) {
+  identify_ns_.nsze = 0;
+  identify_ns_.flbas.format = 0;
+  identify_ns_.lbaf[identify_ns_.flbas.format].lbads = 0;
+  EXPECT_EQ(translator::StatusCode::kSuccess,
+            translator::ReadCapacity10ToScsi(buffer_, nvme_cmds_[0]));
+  scsi::ReadCapacity10Data result = {};
+  EXPECT_TRUE(translator::ReadValue(buffer_, result));
```

Use `ASSERT` rather than `EXPECT` when it doesn't make sense to continue the test after a failure, as is the case here and in other uses of `ReadValue` below.